src/main/java/seedu/address/logic/commands/DeleteMultipleIndexCommand.java

```diff
@@ -0,0 +1,66 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+/**
+ * Deletes a person identified using it's displayed index from the address book.
+ */
+public class DeleteMultipleIndexCommand extends DeleteCommand {
+
+    public static final String MESSAGE_DELETE_PERSON_SUCCESS = "Deleted People Successfully ";
+
+    private final ArrayList<Index> indexes;
+
+    public DeleteMultipleIndexCommand(ArrayList<Index> indexes) {
+        this.indexes = indexes;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Person> lastShownList = model.getFilteredPersonList();
+        isIndexesValid(lastShownList);
+
+        ArrayList<Person> listOfPeople = new ArrayList<Person>();
+        for (Index targetIndex : this.indexes) {
+
+            Person personToDelete = lastShownList.get(targetIndex.getZeroBased());
+            listOfPeople.add(personToDelete);
+        }
+        for (Person personToDelete : listOfPeople) {
+            model.deletePerson(personToDelete);
```

Comment:
As the undoManager saves the state of the addressBook after every deletePerson() method call, this implementation of multiple delete causes undo to restore one person at a time.

Instead, I suggest a new method, `deleteMultiplePersons(List<Person>)` be added to the modelManager, that saves the addressBook state once, after all the deletions have completed.

Reply:
Have just done so. You can take a look at it if you want. Apologies for overlooking this.